pkg/yang/yang.go

```diff
@@ -626,13 +626,13 @@ type Uses struct {
 	Parent     Node         `yang:"Parent,nomerge" json:"-"`
 	Extensions []*Statement `yang:"Ext" json:"-"`
 
-	Augment     *Augment  `yang:"augment" json:",omitempty"`
-	Description *Value    `yang:"description" json:",omitempty"`
-	IfFeature   []*Value  `yang:"if-feature" json:"-"`
-	Refine      []*Refine `yang:"refine" json:"-"`
-	Reference   *Value    `yang:"reference" json:"-"`
-	Status      *Value    `yang:"status" json:"-"`
-	When        *Value    `yang:"when" json:",omitempty"`
+	Augment     []*Augment `yang:"augment" json:"-"`
```

Comment:
This change to an exported field makes your PR not-backwards compatible with folks currently using this field with previous `goyang` releases - we'd prefer to avoid this outcome.

What we'd discussed (sorry, the PR/issue number with the discussion escapes me right now) was to instead have this:

```go
  Augment *Augment `yang:"augment" ...` // for 0th "augment"
  Augments []*Augment `...` // for 1..Nth "augment"
```

Adding the s-suffixed field, implying plurality, stores only the 1..Nth such statement, facilitating backwards compatible use.

To support the above will require some changes to the tag handling code (e.g., here https://github.com/openconfig/goyang/blob/b6e1e28273ad274f9240361b3a1ae65d27c045ac/pkg/yang/node.go#L245), e.g., using a whitelist of field names which are "plural" fields, or changes to the tag structure itself.  The former of those (statically describing plural field literals), though not ideal, seems preferable to me.  My reasoning is that tag reflection has some magic (until you're familiar with both how struct tagging and reflection works and read the specific code in goyang which deals with its tags), while a `switch` statement for the static plural cases is obvious.

Any chance you could give the above approach a try and see if there's any big roadblocks?

Reply:
Oh, thanks @LeonGWang (seeing your comment from three weeks back) - that is the issue in question.

Reply:
@andaru I see, but I had originally pushed this because I wanted the Yang to parse, whereas the codegen with the parsed Yang model was not using these Augments at all (within the Uses block) - was a separate issue to fix.

Could we instead opt to keep Augment == Augments[0] ? that way, you could do both, and going through augments does not involve a custom iteration / chain.

Reply:
There is not any use of the 0, 1..N recipe presently.

Yes, we could have the first `augment` be stored in both `Augment` and `Augments[0]`.  That maintains backward compatibility perfectly.  While it may not be a concern of some users, some users are concerned about memory consumption (see PR 95 for example), so I didn't originally suggest it, but I think it's reasonable.

A third approach (briefly mentioned towards the end of issue 51) would avoid the duplicate storage by adding a method to (each) type.

```go
func (o *Uses) GetAugment() []*Augment { ... }
```

This does (as I understand it) add to the static object/binary size, however, so it's not a clear winner either.

Any comments (from yourself or others, eg @LeonGWang, @robshakir ?) regarding the above?

Reply:
It looks like a change got merged which used `Augment []*Augment` instead of separating the fields out.  Would you mind closing this PR or splitting out any other work you needed into this or a separate PR?

Reply:
I'm glad this is resolved now. :-) Thank you for the support and getting the Yang functionality in.

@andaru @LeonGWang